File: LLM-Translations/GPT-4/avatar/Java/Go/codeforces_445_A.go

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	inp := strings.Split(scanner.Text(), " ")
	n := inp[0]
	m := inp[1]
	var ans []string

	for i := 0; i < n; i++ {
		scanner.Scan()
		str := scanner.Text()
		var temp strings.Builder

		for j := 0; j < m; j++ {
			if str[j] == '-' {
				temp.WriteString("-")
			} else {
				if (i+j)%2 == 1 {
					temp.WriteString("W")
				} else {
					temp.WriteString("B")
				}
			}
		}

		ans = append(ans, temp.String())
	}

	for i := 0; i < n; i++ {
		fmt.Println(ans[i])
	}
}
// //End of Code

Please note that, unlike Java, Go does not throw exceptions, so error handling is done differently. You might need to add error checks where you think it's necessary. The above code assumes that the input will always be correct.

File: LLM-Translations/GPT-4/avatar/Java/Python/codeforces_445_A.py
Here is your Java program translated to Python:


import sys

n, m = map(int, input().strip().split())
ans = []

for i in range(n):
    str = input().strip()
    temp = ""

    for j in range(m):
        if str[j] == '-':
            temp += '-'
        else:
            if (i + j) % 2 == 1:
                temp += 'W'
            else:
                temp += 'B'
    ans.append(temp)

for i in range(n):
    print(ans[i])

##End of Code.
